lib/iris/aux_factory.py

```diff
@@ -188,7 +189,7 @@ def _nd_bounds(coord, dims, ndim):
         # Transpose to be consistent with the Cube.
         sorted_pairs = sorted(enumerate(dims), key=lambda pair: pair[1])
         transpose_order = [pair[0] for pair in sorted_pairs] + [len(dims)]
-        bounds = coord.core_bounds()
+        bounds = coord.lazy_bounds()
         if dims:
```

Comment:
@pp-mo There is no equivalent check here as per [line 221](https://github.com/SciTools/iris/pull/2604/files#diff-42d68d7f5fd99ece9d3d591cd777d86cR221) to check for a `nop` transpose, given an increasing `dims` order ... is it relevant to add that here also?